pinot-controller/src/main/java/org/apache/pinot/controller/BaseControllerStarter.java

```diff
@@ -243,7 +244,7 @@ public void init(PinotConfiguration pinotConfiguration)
       _adminApp = new ControllerAdminApiApplication(_config);
       // Do not use this before the invocation of {@link PinotHelixResourceManager::start()}, which happens in {@link
       // ControllerStarter::start()}
-      _helixResourceManager = new PinotHelixResourceManager(_config);
+      _helixResourceManager = PinotHelixResourceManagerFactory.create(_config);
```

Comment:
better add a `protected ... createPinotHelixResourceManager(config)` method in this Starter class, so that the create method can be overridden by subclass of the Starter class to create an object extending PinotHelixResourceManager

Reply:
Closing this PR as it looks like an overkill. Will open a new one.